### PR TITLE
[mobile][photos] display system trash entries on trash page

### PR DESCRIPTION
**Pull request by aspizu**
- **[mobile][photos] simplify getTrashedFiles function**
- **[mobile][photos] list trash from system trash on android in trash**

### CODE REVIEW

File: mobile/apps/photos/lib/ui/viewer/gallery/trash_page.dart [hunk 1]
```diff
@@ -40,13 +39,38 @@ class TrashPage extends StatelessWidget {
...
+          // Insert local only trash entries
+          final localTrashAssets = (await Future.wait(
+            localTrash.map((item) => AssetEntity.fromId(item.localID)),
+          )).whereType<AssetEntity>().toList();
+          for (var i = 0; i < localTrash.length; i++) {
+            final file = TrashFile.fromEnteFile(
+              fileFromAsset('Unknown Folder', localTrashAssets[i]),
+              createdAt: 0,
+              updateAt: 0,
+              deleteBy: localTrash[i].deleteBy,
```
[the remark this answers is not in the document]
Comment (aspizu) [reply mobile/apps/photos/lib/ui/viewer/gallery/trash_page.dart:69]: we will handle this later when adding separate actions for this case.

File: mobile/apps/photos/lib/ui/viewer/gallery/trash_page.dart [hunk 2]
```diff
@@ -40,13 +39,41 @@ class TrashPage extends StatelessWidget {
...
-        return TrashDB.instance.getTrashedFiles(
-          creationStartTime,
-          creationEndTime,
-          limit: limit,
-          asc: asc,
-        );
+      asyncLoader: (creationStartTime, creationEndTime, {limit, asc}) async {
+        final remoteTrash = await TrashDB.instance.getTrashedFiles();
+        if (flagService.internalUser && Platform.isAndroid) {
+          final localTrash = await NativeService.getTrash();
```
Comment (aspizu) [inline mobile/apps/photos/lib/ui/viewer/gallery/trash_page.dart]: **MEDIUM** — If the MediaStore query or method channel fails—for example after media access is revoked or while the provider volume is unavailable—this await throws after `remoteTrash` has already loaded. The gallery’s first load then never installs those valid remote rows, so the trash page can stay loading instead of showing Ente trash. Catch and log failures from the native lookup, merge an empty local list, and add a test where the channel throws while remote trash is non-empty.

File: mobile/apps/photos/lib/ui/viewer/gallery/trash_page.dart [hunk 3]
```diff
@@ -40,13 +39,41 @@ class TrashPage extends StatelessWidget {
...
+              localTrash.removeWhere((item) => item.localID == file.localID);
+            }
+          }
+          // Insert local only trash entries
+          final localTrashAssets = (await Future.wait(
+            localTrash.map((item) => AssetEntity.fromId(item.localID)),
+          )).toList();
+          for (var i = 0; i < localTrash.length; i++) {
+            if (localTrashAssets[i] == null) continue;
+            final file = TrashFile.fromEnteFile(
```
Comment (aspizu) [inline mobile/apps/photos/lib/ui/viewer/gallery/trash_page.dart:65]: **MEDIUM** — This local-only file gets a `localID`, but no `generatedID` or `uploadedFileID`. `SelectedFiles._isMatch` checks only the two missing IDs, so the first long-press selects the row but a second tap cannot find and remove it; a reload can also leave a stale selection that the rebuilt tile does not show. Match on a non-null `localID` as a fallback (or give these rows another stable selection ID), and test select, tap again to deselect, and reload while selected.

File: mobile/apps/photos/lib/db/trash_db.dart [hunk 4]
```diff
@@ -212,29 +213,19 @@ class TrashDB {
...
     final results = await db.query(
       tableName,
-      where:
-          '$columnTrashDeleteBy != -1 AND '
-          '$columnCreationTime >= ? AND $columnCreationTime <= ?',
-      whereArgs: [startTime, endTime],
-      orderBy: '$columnCreationTime ' + order,
-      limit: limit,
+      where: '$columnTrashDeleteBy != -1',
+      orderBy: '$columnTrashDeleteBy DESC',
```
Comment (aspizu) [inline mobile/apps/photos/lib/db/trash_db.dart:221]: **MEDIUM** — Ordering by `deleteBy` breaks the gallery’s capture-time grouping contract. `GalleryGroups` splits adjacent rows and searches group ranges by `creationTime`; expiry order and capture order are unrelated, so the same day or month can appear under several out-of-order headers and time-based scroll lookup can miss its group. Keep the merged list ordered by `creationTime`, or change/disable capture-time grouping for this page, and test rows whose expiry order is the reverse of their capture order.

**Review by aspizu — COMMENTED** [review]
Automated code review: high- and medium-severity findings only.
